test/unit/test_validate.py

```diff
@@ -0,0 +1,135 @@
+# pylint: disable=C0116
+# pylint: disable=C0114
+from pathlib import Path
+from unittest.mock import Mock
+import warnings
+import mock
+import pytest
+
+from validate.validate import (
+    detect_file_type_and_extension,
+    check_extension
+)
+from validate.files import (
+    check_compressed,
+    path_exists
+)
+from validate.validators.bam import (
+    validate_bam_file,
+    check_bam_index
+)
+from validate.validators.vcf import (
+    validate_vcf_file
+)
+
+def test__detect_file_type_and_extension__detects_correct_file_type():
+    test_path = Path('a.vcf.gz')
+    expected_file_type = 'file-vcf'
+    expected_extensions = ['.vcf', '.gz']
+
+    file_type, extension = detect_file_type_and_extension(test_path)
+
+    assert extension == ''.join(expected_extensions)
+    assert file_type == expected_file_type
```

Comment:
Consider consolidated some parts of this: expected_extensions can be a string; test_path can use the same string; the string can be used in the assertion.